collector/meminfo_numa_linux.go

```diff
@@ -127,8 +149,34 @@ func parseMemInfoNuma(r io.Reader) (map[meminfoKey]float64, error) {
 
 		// Active(anon) -> Active_anon
 		metric = re.ReplaceAllString(metric, "_${1}")
-		memInfo[meminfoKey{metric, parts[1]}] = fv
+		memInfo = append(memInfo, meminfoMetric{metric, prometheus.GaugeValue, parts[1], fv})
 	}
 
 	return memInfo, nil
 }
+
+func parseMemInfoNumaStat(r io.Reader, nodeNumber string) ([]meminfoMetric, error) {
+	var (
+		numaStat []meminfoMetric
+		scanner  = bufio.NewScanner(r)
+	)
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" {
+			continue
+		}
+		parts := strings.Fields(string(line))
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("line scan did not return 2 fields: %s", line)
+		}
+
+		fv, err := strconv.ParseFloat(parts[1], 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid value in numastat: %s", err)
+		}
+
+		numaStat = append(numaStat, meminfoMetric{parts[0], prometheus.CounterValue, nodeNumber, fv})
```

Comment:
Yes, good idea.  The easy way of `+ "_total"` may return some less optimal metric names like `...numa_hit_total` instead of `...numa_hits_total`.  I guess I could add a mapping lookup table for the various stat names to metric names. :-/